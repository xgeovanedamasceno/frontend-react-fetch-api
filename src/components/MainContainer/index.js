import MainContainerStyle from './styled';

function MainContainer() {
  return (
    <MainContainerStyle><main /></MainContainerStyle>
  );
}

export default MainContainer;
