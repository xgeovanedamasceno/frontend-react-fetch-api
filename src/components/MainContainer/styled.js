import styled from 'styled-components';

const MainContainerStyle = styled.div`
  background-color: aquamarine;;
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: 750px;

`;

export default MainContainerStyle;
