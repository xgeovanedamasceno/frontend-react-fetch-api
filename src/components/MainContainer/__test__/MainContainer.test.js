import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import MainContainer from '..';

describe('MainContainer', () => {
  it('should contain the main element when rendering', () => {
    render(<MainContainer />);

    const mainEl = screen.getByRole('main');

    expect(mainEl).toBeInTheDocument();
  });

  it('should render correctely', () => {
    const tree = renderer.create(<MainContainer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
