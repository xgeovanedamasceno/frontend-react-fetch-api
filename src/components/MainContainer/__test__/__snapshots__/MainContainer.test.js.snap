// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer should render correctely 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0 auto;
  width: 750px;
}

<div
  className="c0"
>
  <main />
</div>
`;
